docs/CppInteroperability/Roadmap.md

```diff
@@ -0,0 +1,264 @@
+# Swift and C++ Interoperability Roadmap
+
+## Introduction
+
+**This roadmap is** a sketch, rather than a final design for the “forward” (using C++ APIs from Swift) half of C++ and Swift interoperability. This roadmap outlines some high-level topics related to C++ interoperability, overarching goals that drive the project’s design decisions, a process for evolving C++ interoperability over time, and a collection of specific problems and their potential solutions. 
+
+“Reverse” interoperability (using Swift APIs from C++) is another extremely important part of the interoperability story, however, reverse interoperability has largely different goals and constraints, which necessarily mean a different design and roadmap. Alex Lorenz will be putting together a roadmap for reverse interoperability which outlines these goals and will shed light onto the differences between the two halves of the projects. In the future this roadmap will need to be updated to explain how these two halves fit together and present the “bi-directional” interoperability story that is C++ and Swift interoperability as a whole. 
+
+**C and Objective-C interoperability:** C++ interoperability will have to build off of the previous art of C and Objective-C interoperability, not only to preserve existing Swift codebases and provide a incremental story, but also to reduce unnecessary cognitive burden on developers. Starting with the name, C++ inherits much of C, which means C++ interoperability has a great starting place to build off of. C is a simple language without many high level constructs, so importing C APIs is, relatively, simple. On the other hand, Objective-C adds some high level constructs, such as reference types and a generics model. However, unlike C++, Objective-C has strong idioms; there are only so many patterns for doing some specific thing. So, while C and Objective-C interoperability is a good starting place, the model for importing these languages cannot be taken wholesale for C++ interoperability. 
+
+## Goals
```

Comment:
In my opinion, would be great to list goals clearly as a section header and then expand more. That way it's possible to quickly glance all of the goals at once and focus on the ones the reader most interested in.

E.g.:

**- Safety**

Safety is a top priority for the Swift programming language. Therefore, the Swift compiler should not import APIs that will likely be less safe in Swift than they are in C++ and interop should strive to make imported APIs *safer* in Swift than they are in C++ by....


**- Diagnostics Support**
The Swift language was built with good diagnostics in mind....

**- Performance**
In a similar vein, the Swift programming language strives ...


Just a stylistic comment, I agree with the spirit.